halo2_proofs/src/dev.rs

```diff
@@ -997,6 +998,104 @@ impl<F: FromUniformBytes<64> + Ord> MockProver<F> {
                         .collect::<Vec<_>>()
                 });
 
+        let shuffle_errors =
+            self.cs
+                .shuffles
+                .iter()
+                .enumerate()
+                .flat_map(|(shuffle_index, shuffle)| {
+                    let load = |expression: &Expression<F>, row| {
```

Comment:
Nitpick: I think this `load` is equal to the one used in for the lookups.  Maybe it can be rewriten as single function/closure that takes `expression`, `row` and `self: MockProver<F>`